src/react/useInfiniteQuery.ts

```diff
@@ -34,15 +37,17 @@ export function useInfiniteQuery<
       TError,
       TData,
       TQueryFnData,
+      SData,
       TQueryKey
     >,
     'queryKey'
   >
-): UseInfiniteQueryResult<TData, TError>
+): UseInfiniteQueryResult<TData, TError, SData>
 export function useInfiniteQuery<
   TQueryFnData = unknown,
   TError = unknown,
   TData = TQueryFnData,
+  SData = unknown,
```

Comment:
shouldn't `SData` default to `TData` to be backwards compatible?

Reply:
SData defaults to an empty object and intersects Tdata.  The issue stems around infinite query types passing TData then wrapping it in InfiniteData when it calls the standard query observer type. 

That is how it is passed to the select and how it identifies TData with the select fn. 

So by adding the Sdata type and intersecting it as a second type param, typescript is able to infer that the select fn gives SData its value which is the return type of the select fn whereas TData remains the value that it assigns from either the nextParam fn or the async fn (unless you change that via the select which is interesting to me, but ideal result) 

It's actually a fundamental error in the types causing the issue in the first place, and this PR is technically incorrect (select could return a primitive but we require you return the pages/pagesParams values (although they can be changed) or a type error occurs. 

This is because of the underlying issue around how infinite query ties to query. 

In a deeper rework of types, it may be possible to resolve this, but it would have been a huge changeset from what I can tell. 

This at least provides a framework for Typescript users to be able to transform the infinite data...